WoWsShipBuilder.ViewModels/Other/ShipComparisonViewModel.cs

```
@@ -0,0 +1,762 @@
+
+    public async Task ToggleCategorySelection(ShipCategory value)
+    {
+        if (SelectedCategories.Contains(value))
+        {
+            SelectedCategories.Remove(value);
+        }
+        else
+        {
+            SelectedCategories.Add(value);
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleAllTiers(bool activationState)
+    {
+        SelectedTiers.Clear();
+        if (activationState)
+        {
+            SelectedTiers.AddRange(Enumerable.Range(1, 11));
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleAllClasses(bool activationState)
+    {
+        SelectedClasses.Clear();
+        if (activationState)
+        {
+            SelectedClasses.AddRange(Enum.GetValues<ShipClass>().Except(new[] { ShipClass.Auxiliary }));
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleAllNations(bool activationState)
+    {
+        SelectedNations.Clear();
+        if (activationState)
+        {
+            SelectedNations.AddRange(Enum.GetValues<Nation>().Except(new[] { Nation.Common }));
+        }
+
+        await ApplyFilters();
+    }
+
+    public async Task ToggleAllCategories(bool activationState)
+    {
+        SelectedCategories.Clear();
+        if (activationState)
+        {
+            SelectedCategories.AddRange(Enum.GetValues<ShipCategory>().Except(new[] { ShipCategory.Disabled, ShipCategory.Clan }));
+        }
+
+        await ApplyFilters();
+    }
+
+    public List<ShipComparisonDataWrapper> GetShipsToBeDisplayedList(bool disableHideShipsIfNoSelectedSection = false)
+    {
+        List<ShipComparisonDataWrapper> list = ShowPinnedShipsOnly ? PinnedShipList : FilteredShipList;
+
+        if (!disableHideShipsIfNoSelectedSection)
+        {
+            list = HideShipsIfNoSelectedSection(list);
+        }
+
+        return list;
+    }
+
+    public void EditBuilds(List<ShipComparisonDataWrapper> newWrappers, bool clearCache = false)
```

Consider removing default parameter as it may hide internal logic

---

the clearCache option added here should never be used. It was added because neccessary only once for a specific operation